### ISSUE TITLE
feat: expose per-token log-probabilities for evaluation, debugging, and logit-based scoring

### ISSUE THREAD

**Issue by chaofengw-nv** [NVIDIA/TensorRT-Model-Connect#248]
## Problem

`TextResult` currently returns only `text` and `token_ids`. The per-token log-probability is computed inside `ISampler::sample()` — `SampleResult` already carries a `logprob` field (marked *informational*) — but it is discarded before reaching the caller.

This blocks several important use cases:

### 1. Benchmark evaluation (MMLU, HellaSwag, etc.)
Standard LM evaluation frameworks (LM-Eval-Harness, EleutherAI) score tasks via `loglikelihood(context, continuation)`. Without API-level logprob access, `task_eval.py` currently works around this in two ways:
- **Letter-cloze**: uses the internal `text_trace` debug facility to dump top-k logits to a JSONL file, then reads it back in Python — a file-I/O detour, not an API call.
- **Full sequence log-prob**: falls back to a separate HF subprocess (`run-hf-logprob`), meaning the `.trtfb` bundle is **not used for this path at all**.

### 2. Debugging model quality
When output is wrong, the first diagnostic question is "was the correct answer in the top-k?" — impossible without logprobs. Today this requires the internal `text_trace` tool, which is not a public interface.

### 3. Confidence scores / uncertainty
Applications that need to gate on model confidence (RAG reranking, abstention, structured output verification) have no signal to work with.

### 4. Perplexity measurement
Not currently possible through the trtmc API.

## What already exists

- `SampleResult::logprob` in `include/trtmc/runtime/sampler.h` — computed on every step, but not collected into `TextResult`.
- `text_trace` config facility — dumps top-k logits to a file; used as a workaround in `task_eval.py::run_trtfb_trace_cloze`.

## API compatibility concern

Naively adding fields to `TextResult` or `GenerateConfig` is **source-compatible** (existing callers recompile cleanly) but **binary-incompatible**: struct layout changes break any pre-compiled `.so` or wheel that was linked against the old header. `GenerateConfig` is already large (~20 fields), so further growth also increases maintenance burden.

### Option A — New virtual methods on `IPipeline` (recommended minimal change)

Add two new optional methods with default `throw` implementations, matching the existing pattern in `IPipeline`:

```cpp
// Compute log P(continuation | context). Returns sum of per-token log-probs.
// Default impl throws; only TextGenerationPipeline overrides it.
virtual float loglikelihood(const std::string& context, const std::string& continuation) {
    throw std::runtime_error(std::string(pipeline_type()) + " does not support loglikelihood()");
}

// Score a single next-token distribution: returns log-prob of `token_id` given `context`.
virtual float score_token(const std::string& context, int32_t token_id) {
    throw std::runtime_error(std::string(pipeline_type()) + " does not support score_token()");
}
```

- **Zero impact on `TextResult` / `GenerateConfig` layout** — existing binaries are unaffected.
- Adding virtual methods does change the vtable, so pre-compiled dynamic libraries still need a rebuild — but that is the standard expectation for minor version bumps.
- Implementation cost is low: reuses the existing `run_step()` loop; just accumulates `SampleResult::logprob` instead of discarding it.
- Directly enables LM-Eval-Harness integration and replaces the HF-subprocess fallback in `task_eval.py`.

### Option B — Opt-in fields in `TextResult` + `GenerateConfig`

```cpp
// GenerateConfig addition (opt-in, zero cost when false)
bool return_logprobs{false};
int32_t return_top_k_logprobs{0};  // 0 = selected token only

// TextResult addition
struct TokenLogprob { int32_t token_id; float logprob; };
std::vector<TokenLogprob> logprobs;             // one per generated token
std::vector<std::vector<TokenLogprob>> top_logprobs;  // top-k per step
```

Mirrors OpenAI Chat Completions (`logprobs`, `top_logprobs`) and HuggingFace `generate(output_scores=True)`. Source-compatible addition, but changes struct layout — callers must recompile.

### Option C — Promote `text_trace` to a first-class config schema

`text_trace` already works; formalizing it (stable schema, documented output format, guaranteed top-k coverage for the answer tokens) would legitimize the current `task_eval.py` workaround without any API surface change. Lower capability ceiling than A/B, but zero risk.

## Recommended path

**Option A first** (new virtual methods) — unblocks evaluation and perplexity with no struct-layout risk. **Option B** as a follow-on for richer per-token data when the team is ready to bump the minor version and rebuild dependents.

## Impact summary

| Use case | Current workaround | With Option A |
|---|---|---|
| MMLU / loglikelihood scoring | HF subprocess (bundle not used) | `loglikelihood()` API |
| Sequence log-prob / perplexity | Not possible via trtmc | `loglikelihood()` API |
| Debug "was correct answer in top-k?" | Internal `text_trace` file hack | `score_token()` + top-k |
| Confidence gating | Not possible | `loglikelihood()` or `score_token()` |
